protocol.md

```diff
@@ -870,6 +902,12 @@ export interface TextDocumentClientCapabilities {
 ```typescript
 interface ClientCapabilities {
 	/**
+	 * The client supports receiving the result solely through $/partialResult notifications for requests from the client to the server.
+	 * If true, the server should send all data through $/partialResult and send null as the result in the final response.
```

Comment:
You are right we can just translate it at the protocol layer, but something about that doesn't feel right. So essentially when streaming is enabled, responses become nothing more than markers that we are done responding.

I worry about jsonrpc2 implementations making it difficult to construct a consistent order of when it receives a notification and a response, since we are now dictating an ordering requirement on responses/notifications that never existed before.